File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    i = 0
    time = 0
    while i < len(progresses):
        
        cal = []
        while i < len(progresses):
            if progresses[i] + time*speeds[i] >= 100:
                cal.append(progresses[i])
                i += 1
            else : break
        if len(cal) != 0:
            answer.append(len(cal))
        time += 1
    
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;
import java.io.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        
        List<Integer> ans = new ArrayList<>();
        int pointer = 0;
        int time = 0;
            
        while (pointer < progresses.length){
            
            if ((progresses[pointer] + time*speeds[pointer]) < 100){
                
                time++;
                
            }
            else{
                int count = 0;
                while ((pointer < progresses.length) && (progresses[pointer] + time*speeds[pointer]) >= 100){
                    
                    pointer++;
                    count++;
                    
                }
                ans.add(count);
                
                
            }
            
            
        }
        
        
        return ans.stream().mapToInt(i -> i).toArray();
    }
}